fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/todos')
+      .then(response => response.json())
+      .then(todoList => {
+        this.setState({
+          todos: todoList
+        });
+      });
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(newTodo) })
+      .then(response => response.json())
+      .then(added => {
+        this.setState(prevState => ({
+          todos: prevState.todos.concat(added)
+        }));
+      });
+  }
+
+  toggleCompleted(todoId) {
+    let todoIndex;
+    const copyArr = [...this.state.todos];
+
+    for (let i = 0; i < this.state.todos.length; i++) {
+      if (todoId === this.state.todos[i].todoId) {
+        todoIndex = i;
+      }
+    }
+
+    const newObj = copyArr[todoIndex];
```

Comment:
This isn't a new object, it's the original object, since objects (and arrays, functions) are pass-by-reference. Make a new object, like you normally would.